decidim-budgets/app/controllers/decidim/budgets/admin/budgets_controller.rb

```diff
@@ -52,32 +53,26 @@ def update
           end
         end
 
-        def destroy
-          enforce_permission_to(:delete, :budget, budget:)
-
-          DestroyBudget.call(budget, current_user) do
-            on(:ok) do
-              flash[:notice] = I18n.t("budgets.destroy.success", scope: "decidim.budgets.admin")
-            end
-
-            on(:invalid) do
-              flash.now[:alert] = I18n.t("budgets.destroy.invalid", scope: "decidim.budgets.admin")
-            end
-          end
+        private
 
-          redirect_to budgets_path
+        def trashable_deleted_resource_type
+          :budget
         end
 
-        private
+        def trashable_deleted_collection
+          @trashable_deleted_collection ||= Budget.where(component: current_component).trashed.deleted_at_desc
+        end
 
         def budgets
-          @budgets ||= Budget.where(component: current_component).order(weight: :asc)
+          @budgets ||= Budget.where(component: current_component).not_trashed.order(weight: :asc)
         end
 
         def budget
-          @budget ||= budgets.find_by(id: params[:id])
+          @budget ||= Budget.where(component: current_component).find_by(id: params[:id])
```

Comment:
I have seen that in accountability you take into account the `not_trashed` state. Is there any reason why you ignore this here ?
```suggestion
          @budget ||= budgets.find_by(id: params[:id])
```